### PR TITLE
New setCustomModal API

### PR DESCRIPTION
**Pull request by jniemin**
New setCustomModal API which allows users to create custom modals

### CODE REVIEW

File: src/redux/selectors/exposedSelectors.js [hunk 1]
```diff
@@ -170,6 +172,8 @@ export const getCursorOverlayData = state => state.viewer.cursorOverlay;
 
 export const getOpenElements = state => state.viewer.openElements;
 
+export const getDisabledElements = state => state.viewer.disabledElements;
```
Comment (jniemin) [inline src/redux/selectors/exposedSelectors.js:175]: As user is able to add more than one custom modal and they are rendered all at the same time, I needed to expose all disabledElements. Previously we had redux selector only to allow predefined modals one by one.

File: src/apis/setCustomModal.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
+import actions from 'actions';
+
+function generateSetCustomModal(store) {
+
+  /**
+   * Add custom modal element to WebViewer.
```
Comment (jniemin) [inline src/apis/setCustomModal.js:6]: How dow we handle versioning on our documentation? For now this is only for 7.0

**Comment by ZhijieZhang** [comment]
@khein-pdftron Can you review it when you have time? Thanks.

**Comment by jniemin** [comment]
> I tried the sample code in the documentation. I couldn't close the modal by clicking outside of it. Can we make that the default behaviour? All other modals do that.

I'll add that. I will make that on by default and possibility to opt-out from it using options

**Comment by jniemin** [comment]
@khein-pdftron Custom modal now supports automatic closing with option to opt-out

**Pull request merged**